src/idlcxx/src/streamers.c

```diff
@@ -152,48 +152,90 @@ process_case(
 {
   struct streams *streams = user_data;
   const idl_case_t* _case = (const idl_case_t*)node;
-  char *type = NULL;
+  char *type, *value = "";
   const char *name;
+  bool simple, single;
 
   static const char writefmt[] =
```

Comment:
maybe replace writefmt and movefmt with:
    "      %s(str, instance.%s());\n"
    "      break;\n";

and then in the print add "write" for the write stream and "move" for the move stream?